### PR TITLE
[Inductor][FlexAttention] Respect user's input kernel_options

### PR DESCRIPTION
**Pull request by yanboliang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #134065

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: test/inductor/test_flex_attention.py [hunk 1]
```diff
@@ -1465,6 +1465,23 @@ def mask_mod(b, h, q, kv):
...
+            dtype=torch.float32,
+            requires_grad=True,
+        )
+        q, k, v = make_tensor(), make_tensor(), make_tensor()
+
+        # Ensure we respect user's input kernel options.
+        _, code = run_and_get_code(
+            torch.compile(flex_attention), q, k, v, kernel_options={"BLOCK_M": 64}
+        )
+        FileCheck().check("BLOCK_M : tl.constexpr = 64").run(code[1])
```
Comment (Chillee) [inline test/inductor/test_flex_attention.py:1483]: Choose a smaller value that we wouldn't normally use like 16.

**Review by Chillee — APPROVED** [review]

**Comment by yanboliang** [comment]
CI failure is from a dynamo missing feature, #134083 can fix it. Will rebase this after that PR got merged.

**Pull request closed without merging**